### ISSUE TITLE
pass parameter to quast

### ISSUE THREAD

**Issue by abrozzi** [TGAC/RAMPART#29]
Is it possible to specify in the xml file this option for quast?

--min-contig (or -m) <int>
Lower threshold for a contig length. Shorter contigs won't be taken into account (except for some metrics, see section 3). The default value is 500.

thanks a lot!

**Comment by maplesond** [comment]
Currently no.  For MECQ and MASS jobs there is either the checked_args or unchecked_args options for specifying additional arguments to the tools but this does not cover anything in the mass analysis stage which include Quast.  Would you like this added as a feature to RAMPART?

**Comment by abrozzi** [comment]
Yes, if it is possible to extend the checked_args or unchecked_args options also to mass. At least for quast.
For some samples rampart was giving an error. I saw in fact that the contigs were ok but only short (slightly less than 500 bps which is the default threshold for QUAST). Changing that parameter --min-contig will solve the problem.

**Comment by maplesond** [comment]
I've pushed a change to the develop branch.  This should allow you to add unchecked_args as an attribute to the Quast tool in the XML file.  e.g.:

`<tool name="QUAST" threads="8" memory="10000" unchecked_args="--min-contig=400"/>`

HTH

**Comment by MaestSi** [comment]
Dear Daniel,
thank you for this modification.
I am now able to run it, and it accepts the "--min-contig" option in the .xml file without giving any errors.
Despite this, the option isn't really added to quast.py command while running.

```
2016-06-29 12:19:13 INFO  DefaultProcessService:146 - Running command in foreground 
[quast.py --output=/home/.../rampart_out/5_mass_analysis/quast/contigs --threads=20 --est-ref-size=2000000 --gene-finding /home/.../rampart_out/4_mass/spades/contigs/spades-cvg-5-contigs.fa].
```

If it isn't too time demanding, could you please fix it?
Thank you.

**Comment by maplesond** [comment]
Could you send me the part of the xml where you are calling Quast?  I can't see from the code why this wouldn't be added at the moment.

**Comment by maplesond** [comment]
Not to worry.  I've managed to reproduce the problem.  I'll push up a fix shortly.

**Comment by maplesond** [comment]
Ok, I've updated the tgac-conan-procs and rampart projects.  If you could pull changes from both (develop branches), then compile tgac-conan-procs first, then rampart second and see if that works.

The issue, if your interested, was that previously I was overriding the way quast generates command lines from the default behaviour, hence it wasn't including the unchecked args.

**Comment by MaestSi** [comment]
Thanks, now everything works!

**Comment by abrozzi** [comment]
Hi Daniel,

in the **weightings.tab** I understand that the **number of genes** is calculated by Quast:

`
nb_genes = 0.2              # Number of genes estimated to be in assembly from Quast (requires estimated number of genes)
`

I specified in the xml file the organism's parameters:

`
<organism name="Neisseria meningitidis" ploidy="1">
   <reference name="MC58" path=AE002098.fna"/>
 <estimated est_genome_size="2272360" est_gc_percentage="51.53" est_nb_genes="2104"/>	
</organism>
`

and QUAST parameters with unchecked_args:

` <tool memory="10000" name="QUAST" threads="20" unchecked_args="--min-contig=200 -G=AE002098.gff -R=AE002098.fna"/>  `
	

but I obtain a 0 in Number of genes:


>2016-12-27 16:46:54 INFO  Amp:144 - Reference genome size: 2272360
>2016-12-27 16:46:54 INFO  Amp:145 - Reference GC%: 51.53
>2016-12-27 16:46:54 INFO  Amp:146 - Reference # Genes: 0


and then:

` 
Conservation Metrics:
 - Total assembly size: 2286240
 - Assembly size from sequences > 1KB: 2150764
 - GC%: 51.71
 - Number of genes: 0
` 

Quast also ran without the passed parameters:

` 
2016-12-27 16:46:53 INFO  DefaultProcessService:146 - Running command in foreground  quast.py --output=Sample_ARG-3263/rampart_out/6_mass_select/ref_quast --threads=1 --gene-finding AE002098.fna
` 

What shall I do to use the number of genes in the selecting step?

Thank you very much
